docs/howto/csrf-protection.txt

```diff
@@ -0,0 +1,321 @@
+.. _using-csrf:
+
+===================================
+How to use Django's CSRF protection
+===================================
+
+To take advantage of CSRF protection in your views, follow these steps:
+
+1. The CSRF middleware is activated by default in the :setting:`MIDDLEWARE`
+   setting. If you override that setting, remember that
+   ``'django.middleware.csrf.CsrfViewMiddleware'`` should come before any view
+   middleware that assume that CSRF attacks have been dealt with.
+
+   If you disabled it, which is not recommended, you can use
+   :func:`~django.views.decorators.csrf.csrf_protect` on particular views
+   you want to protect (see below).
+
+2. In any template that uses a POST form, use the :ttag:`csrf_token` tag inside
+   the ``<form>`` element if the form is for an internal URL, e.g.:
+
+   .. code-block:: html+django
+
+       <form action="" method="post">{% csrf_token %}
+
+   This should not be done for POST forms that target external URLs, since
+   that would cause the CSRF token to be leaked, leading to a vulnerability.
+
+3. In the corresponding view functions, ensure that
+   :class:`~django.template.RequestContext` is used to render the response so
+   that ``{% csrf_token %}`` will work properly. If you're using the
+   :func:`~django.shortcuts.render` function, generic views, or contrib apps,
+   you are covered already since these all use ``RequestContext``.
+
+.. _csrf-ajax:
+
+Using CSRF protection with AJAX
+===============================
+
+While the above method can be used for AJAX POST requests, it has some
+inconveniences: you have to remember to pass the CSRF token in as POST data with
+every POST request. For this reason, there is an alternative method: on each
+XMLHttpRequest, set a custom ``X-CSRFToken`` header to the value of the CSRF
+token. This is often easier, because many JavaScript frameworks provide hooks
+that allow headers to be set on every request.
+
+First, you must get the CSRF token. How to do that depends on whether or not
+the :setting:`CSRF_USE_SESSIONS` setting is enabled.
+
+Acquiring the token if :setting:`CSRF_USE_SESSIONS` is ``False``
+----------------------------------------------------------------
+
+The recommended source for the token is the ``csrftoken`` cookie, which will be
+set if you've enabled CSRF protection for your views as outlined above.
+
+.. note::
+
+    The CSRF token cookie is named ``csrftoken`` by default, but you can control
+    the cookie name via the :setting:`CSRF_COOKIE_NAME` setting.
+
+    The CSRF header name is ``HTTP_X_CSRFTOKEN`` by default, but you can
```

Comment:
This is the transformed variant (ie what ends up in the wsgi environ), the actual name is `X-CSRFToken`

Reply:
So the user could set ``CSRF_HEADER_NAME`` to ``X_MY_CSRFTOKEN`` and this would become ``HTTP_X_MY_CSRFTOKEN`` in the WSGI environment?

Reply:
The user would set `HTTP_X_MY_CSRFTOKEN` in the settings but send `X-My-CSRFtoken` in the javascript